### PR TITLE
Synthflesh can heal dead people

### PR DESCRIPTION
**Pull request by XDTM**
This is mostly to make Strange Reagent usable; right now it's pointless unless people are dying to toxin damage.

### CODE REVIEW

**Comment by Ergovisavi** [comment]
We had patches act like this previously and it made defibs ridiculously strong.

That said, in a world where anyone can decapitate a mob, or remove organs to render a defib useless, I wonder if that's too bad of a thing. However, defibs having a time limit and strange reagent not makes me err on the side of not wanting this. If strange reagent was just a defib under another name, it'd be fine, but it having an infinite timer isn't great.

:-1:

**Comment by XDTM** [comment]
It is rarely if ever made, due to the hard to obtain reagents that make it; for that difficulty in making it there is little to no reward, since 100 damage is what you need to end up in crit, let alone die; unless the victim died by poison or oxyloss, the only way you'll revive someone with it is if he died by taking equal burn and brute damage in quantities less than 100, which is, well, rare.

**Comment by kevinz000** [comment]
Fun Fact: Medical beamguns heal the dead. Do your R&D and mount it on a mech for funsies.

**Comment by Anonmare** [comment]
It's not as big a buff to defibs anymore considering defibs will only work 2 minutes after someone's death. You'd need some damn quick response times to get to the patient, synth flesh them up if they're DOA and get the paddles ready to shock them.

**Comment by FantasticFwoosh** [comment]
If you could ready up a body for brain transplant with synthflesh using this theoretically. Frankenstienian human hybrid horror made out of corpses and revived with the power of science when?

Already with the route that the addition limb grower for expanding the use of synthflesh is going it certainly looks that way

**Comment by Iamgoofball** [comment]
:+1:

**Comment by Xhuis** [comment]
:+1: F :+1: O :+1: A :+1: M :+1: E :+1: S :+1: T :+1: R :+1: Y :+1:

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -331,12 +331,13 @@
...
 
 /datum/reagent/medicine/synthflesh/reaction_mob(mob/living/M, method=TOUCH, reac_volume,show_message = 1)
-	if(iscarbon(M) && M.stat != DEAD)
-		if(method in list(PATCH, TOUCH))
-			M.adjustBruteLoss(-1.25 * reac_volume)
-			M.adjustFireLoss(-1.25 * reac_volume)
-			if(show_message)
-				M << "<span class='danger'>You feel your burns and bruises healing! It stings like hell!</span>"
+	if(iscarbon(M) && M.stat == DEAD)
+		show_message = 0
```
Comment (phil235) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: you see the message when you're dead? "it stings like hell" -> feeling pain when dead.

Comment (phil235) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: so dead simple animals will still get the message?

**Comment by Iamgoofball** [comment]
strange reagent was supposed to be a riskier definitely with a gib on fail and no time limit

Then you cocksuckers removed the gib

**Comment by Iamgoofball** [comment]
*defib

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (XDTM) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Fixed

**Comment by coiax** [comment]
Aside, do support guardians still heal the dead? I remember that was a way of getting people back alive, back when we had 10 minute defib windows.

**Comment by XDTM** [comment]
Since holoparasites were removed except as a rare mining drop, it's easier to go with what kevinz000 said and build a mech medigun currently.

**Comment by coiax** [comment]
Hey @KorPhaeron I thought holoparasites got removed for "thematic" reasons. They coming back?

**Comment by optimumtact** [comment]
We should just remove the defib

**Comment by FantasticFwoosh** [comment]
Post - playing with some features of this PR, this is a badly implemented addition with noble but misguided intentions. 

It takes 50+ reagent to heal a badly damaged corpse ineffectively with all the affected parts sucking in sythflesh, and even when your body is healed you still require strange reagent to revive a corpse and to refill with blood/organs if nessecary. What is the point? You can more easily bypass the entire meme process by means of augumentation, healing the entire body is a in-effective when you can fit new mechanical full health parts that are easier to heal.

> Headless corpses can be fed synthflesh in a broken reaction, i would make a issue report but addressing the root cause of the 'feature' would be more apt.

Full health part attachment is better than healing the entire body conventionally (look at aug parts/synthetic limbs etc) and this is where i think this PR has fallen by the wayside. With full health parts attached at worst only immediate toxin damage could kill the strange reagent revived person or put them into critical to be treated further rather than slipping instantly back into death from combined damage or never waking up at all.

There are advantages too, to restoring the owner's own limb back to health and re-attaching since it induces next to no toxin damage whatsoever compared to a foreign donor or synthetic/robotic limb.

> A system to restore individual limb parts to pristine health and/or apply synth-flesh reagent rather than pill/patch like a post-death bruise/burn pack onto targeted body parts would be better and more balanced moving away from old gripes about instant revival patches. 

> You can already get around using synthflesh in the way this PR describes using mech medigun too, especially if complimented with a mech syringe gun full of strange reagent.

**Comment by Iamgoofball** [comment]
just remove meds and replace with augs

**Comment by Iamgoofball** [comment]
Strange Reagent is patch only btw

**Comment by optimumtact** [comment]
I'm not accepting any buffs to this item, but thank you for investigating items that could require a nerf, I will look into them.

**Comment by PKPenguin321** [comment]
@iamgoofball I have gotten strange reagent to work via firing it out of a syringe gun before

**Comment by Iamgoofball** [comment]
@PKPenguin321 thats a bug or some dipshit broke my reagent again

**Pull request merged**